database/useList.ts

```diff
@@ -138,7 +138,11 @@ export default (query: database.Query): ListHook => {
 
   useEffect(
     () => {
-      const query: database.Query = ref.current;
+      const query: database.Query | null | undefined = ref.current;
+      if (!query) {
+        dispatch({ type: 'value' })
```

Comment:
@neilor Currently this means that when no query is specified, `useList` will have an empty array as its value.  Do you think this is correct?  I'm torn as to whether this should be `undefined` or `null` as this better represents an absence of a value.

Reply:
Nice job!